aws-pinpoint-core/src/main/java/com/amplifyframework/pinpoint/core/endpointProfile/EndpointProfile.kt

```diff
@@ -39,14 +41,17 @@ class EndpointProfile(
     uniqueId: String,
     appDetails: AndroidAppDetails,
     deviceDetails: AndroidDeviceDetails,
-    applicationContext: Context
+    applicationContext: Context,
+    private val store: KeyValueRepository
 ) {
     private val attributes: MutableMap<String, List<String>> = ConcurrentHashMap()
     private val metrics: MutableMap<String, Double> = ConcurrentHashMap()
     private val currentNumOfAttributesAndMetrics = AtomicInteger(0)
 
     var channelType: ChannelType? = null
-    var address: String = ""
+    val address: String get() {
+        return store.get(TargetingClient.AWS_PINPOINT_PUSHNOTIFICATIONS_DEVICE_TOKEN_KEY) ?: ""
```

Comment:
I'm not too concerned about this at the moment because it would require a restructuring in how we are storing our data in shared preferences. If we support multiple endpoints, we would need to support the saving of those multiple endpoints, and the code in EndpointProfile would have to be adjusted to support read multiple profiles.

Another point is that a device can only have 1 token. Each endpoint would have the same token. However, the address would only get passed if the endpoint channel was FCM.